js/sts-hotspot.js

```diff
@@ -0,0 +1,72 @@
+AFRAME.registerComponent('sts-hotspot', {
+  schema: {
+    show: { type: 'selector' },
+    sound: { type: 'selector' },
+    playing: { default: false }
+  },
+
+  init: function () {
+    var self = this;
+    var popupEl = this.data.show;
+
+    popupEl.dataset.hotspot = self.el.id;
+
+    self.el.addEventListener('mouseenter', function () {
+      self.el.setAttribute('sts-hotspot', { playing: true });
+      self.el.dataset.playing = true;
+
+      popupEl.setAttribute('material', { visible: true });
+      popupEl.setAttribute('visible', true);
+    });
+
+    window.requestIdleCallback(this.preloadAsset.bind(this));
+  },
+
+  preloadAsset: function () {
+    var sound = this.data.sound;
+    sound.load();
+  },
+
+  stopAllHotspotSound: function () {
+    var AllHotspotSound = document.querySelectorAll('[data-playing]');
```

Comment:
nit: `AllHotspotSound` -> `allHotspotSounds`

Reply:
was my theory correct that multiple sounds could be playing, or was I totally off on that?

I wouldn't spend too much time probing that. I'm getting nitpicky here. asking for my own curiosity.